Fix pylint issues.

Seems `pylint` has been updated and now finds some _new_ issues.
```
************* Module auto_analyse_raw_data
tools/auto_analyse_raw_data.py:727:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module scrape_supported_devices
tools/scrape_supported_devices.py:176:47: R1734: Consider using [] instead of list() (use-list-literal)
```

--- a/tools/auto_analyse_raw_data.py
+++ b/tools/auto_analyse_raw_data.py
@@ -1,9 +1,9 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 """Attempt an automatic analysis of IRremoteESP8266's Raw data output.
    Makes suggestions on key values and tried to break down the message
    into likely chunks."""
 #
-# Copyright 2018 David Conran
+# Copyright 2018-2021 David Conran
 import argparse
 import sys
 
@@ -724,7 +724,7 @@ def get_rawdata(arg_options):
   if arg_options.stdin:
     return sys.stdin.read()
   if arg_options.file:
-    with open(arg_options.file) as input_file:
+    with open(arg_options.file, encoding="utf8") as input_file:
       return input_file.read()
   else:
     return arg_options.rawdata

--- a/tools/scrape_supported_devices.py
+++ b/tools/scrape_supported_devices.py
@@ -173,7 +173,7 @@ def getalldevices():
     protocol = match.group(1)
     for brand, model in supports:
       protocolbrand = (protocol, brand)
-      pbset = sets.allcodes.get(protocolbrand, list())
+      pbset = sets.allcodes.get(protocolbrand, [])
       if model in pbset:
         print("Model %s is duplicated for %s, %s" % (model, protocol, brand))
       sets.allcodes[protocolbrand] = pbset + [model]

